### PR TITLE
[feat]syncing model

### PR DESCRIPTION
**Pull request by Emilypeng1104**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
1. Added new SyncModal component， supports two states: success and fail

2. Integrated modal into SyncButton
    Modal now shows after sync API call:
     -success → shows success modal
     -error → shows fail modal

3. Added SVG icons for Success & Failure
    Imported into lib/icons.tsx
    Includes:
    -SyncSuccess 
    -SyncFail 
    -cross for cancel
4. Implemented full modal styling

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

success: 
fail: 

## 👍 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
1.Start with the UI component
   components/SyncModal/index.tsx
   components/SyncModal/styles.ts
2.Check SVG icons
   lib/icons
3.Review SyncButton integration
   components/SyncButton/index.tsx
4.test locally
   Trigger Sync success
   Trigger Sync error
5.Confirm layout consistency across different screen widths

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @nehaahussain

### CODE REVIEW

File: components/SyncModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        {isSuccess ? (
+          <ActionsRow>
+            <ContinueButton onClick={onClose}>
+              <ContinueText>Continue</ContinueText>
+            </ContinueButton>
+          </ActionsRow>
+        ) : (
+          <ActionsRow>
+            <TryLaterButton onClick={onClose}>Try Later</TryLaterButton>
+            <SyncAgainButton onClick={onClose}>Sync Again</SyncAgainButton>
```
Comment (nehaahussain) [inline components/SyncModal/index.tsx]: when the sync again button is pressed, it should call `handleSync()` again.

File: components/SyncModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+const SyncModal: React.FC<SyncModalProps> = ({ variant, onClose }) => {
+  const isSuccess = variant === "success";
+
+  return (
+    <Overlay onClick={onClose}>
+      <Card onClick={e => e.stopPropagation()}>
+        <CloseButton onClick={onClose}>
+          <Cross />
```
Comment (nehaahussain) [inline components/SyncModal/index.tsx]: same comment as below

File: components/SyncButton/styles.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
+import styled from "styled-components";
```
[the remark this answers is not in the document]
Comment (Emilypeng1104) [reply components/SyncButton/styles.ts]: Deleted
